covid_graphs/covid_graphs/scripts/country_dashboard.py

```diff
@@ -27,248 +27,269 @@ def __str__(self):
+                if country_graph.short_name == country_short_name
+            ]
+            if len(graphs) == 0:
+                return dash.no_update
+
+            figure = graphs[0].create_country_figure(graph_type)
+            return figure
+
+    def _create_all_countries_callbacks(self):
+        dash_graph_dict = {
+            prediction_event_name: [
+                dcc.Graph(
+                    id=f"{graph.short_name}-graph-{prediction_event_name}",
+                    figure=graph.create_country_figure(),
+                    config=dict(modeBarButtons=[["toImage"]]),
+                )
+                for graph in self.graph_dict[prediction_event_name]
+            ]
+            for prediction_event_name in self.prediction_event_by_name.keys()
+        }
 
-    dash_graph_dict = {
-        (prediction_event_name, graph_type.name): [
-            dcc.Graph(
-                id=f"country-graph-{graph.short_name}",
-                figure=graph.create_country_figure(graph_type),
-                config=dict(modeBarButtons=[["toImage"]]),
-            )
-            for graph in graph_dict[prediction_event_name]
+        @self.app.callback(
+            [
+                Output("country-graphs", component_property="children"),
+                Output("graph-title", component_property="children"),
+            ],
+            [Input("prediction-event", "value")],
+        )
+        def update_event(prediction_event_name):
+            graphs = dash_graph_dict[prediction_event_name]
+            next_day = self.prediction_event_by_name[prediction_event_name].date + timedelta(days=1)
+            return graphs, f"{next_day.strftime('%B %d')} predictions"
+
+        # TODO(lukas): only have one callback for all prediction events
+        @self.app.callback(
+            [
+                Output(f"{graph.short_name}-graph-{BK_20200411.name}", component_property="figure")
+                for graph in self.graph_dict[BK_20200411.name]
+            ],
+            [Input("graph-type", "value")],
+        )
+        def update_country_graphs_20200411(graph_type_str):
+            result = []
+            for graph in self.graph_dict[BK_20200411.name]:
+                graph.update_graph_type(GraphType[graph_type_str])
+                result.append(graph.figure)
+            return result
+
+        @self.app.callback(
+            [
+                Output(f"{graph.short_name}-graph-{BK_20200329.name}", component_property="figure")
+                for graph in self.graph_dict[BK_20200329.name]
+            ],
+            [Input("graph-type", "value")],
+        )
+        def update_country_graphs_20200329(graph_type_str):
+            result = []
+            for graph in self.graph_dict[BK_20200329.name]:
+                graph.update_graph_type(GraphType[graph_type_str])
+                result.append(graph.figure)
+            return result
+
+    def _get_header_content(self, title: str):
```

Comment:
It is fine to have functions outside of a class in a file in python :)

Reply:
Done